pkg/kubelet/nodestatus/setters.go

```diff
@@ -160,16 +162,19 @@ func NodeAddress(nodeIP net.IP, // typically Kubelet.nodeIP
 			} else if addr := net.ParseIP(hostname); addr != nil {
 				ipAddr = addr
 			} else {
-				var addrs []net.IP
-				addrs, _ = net.LookupIP(node.Name)
+				ctx, cancel := context.WithTimeout(context.Background(), nodeStatusUpdateFrequency/2)
+				defer cancel()
+
+				addrs, _ := net.DefaultResolver.LookupIPAddr(ctx, hostname)
```

Comment:
Can you share more about why this changed from `hostname` to `node.Name`?

Reply:
Also... its strange that there's no `LookupIP` method that takes a `context`... there's not much we can do about it, but I'm wondering if you know why that is?

Reply:
> Can you share more about why this changed from `hostname` to `node.Name`

thanks, changed to `node.Name`

>  its strange that there's no LookupIP method that takes a context

In my personal opinion, `net.LookupIP`  is just a wrapper of `DefaultResolver.LookupIPAddr` for easy use, if you have some advanced need, you can use `LookupIPAddr` directly, and pass a context into it.

you can find this implementation at here:
https://github.com/golang/go/blob/a6a7b148f874b32a34e833893971b471cd9cdeb7/src/net/lookup.go#L188-L200